proto/iotics/api/search.proto

```diff
@@ -83,6 +92,111 @@ message SearchRequest {
   Range range = 20;
 }
 
+// AdvancedSearchRequest describes a search request with more filtering possibilities than SearchRequest. It returns
+// formatted details about the twins matched by the supplied filter.
+message AdvancedSearchRequest {
+  // Search request payload.
+  message Payload {
+
+    // Expected response type
+    ResponseType responseType = 1;
+
+    // The search filter, expressed as <LANG-TODO>
+    string filter = 2;
+  }
+
+  // Search request headers
+  Headers headers = 1;
+
+  // Search request scope
+  Scope scope = 2;
+
+  // Search request payload
+  Payload payload = 3;
+
+  // Search result limit (optional), affecting the maximum number of twins returned, per host. If not specified, no
+  // limit will be applied.
+  Limit limit = 4;
+}
+
+// AdvancedSearchRawRequest describes a search request with more filtering possibilities than SearchRequest. It returns
+// encoded RDF about the twins matched by the supplied filter.
+message AdvancedSearchRawRequest {
+  // Search request payload.
+  message Payload {
+
+    // Expected response type
+    RdfResultType responseType = 1;
+
+    // The search filter, expressed as <LANG-TODO>
+    string filter = 2;
+  }
+
+  // Search request headers
+  Headers headers = 1;
+
+  // Search request scope
+  Scope scope = 2;
+
+  // Search request payload
+  Payload payload = 3;
+
+  // Search result limit (optional), affecting the maximum number of twins returned, per host. If not specified, no
```

Comment:
(Limit to be discussed)

Reply:
@StephanieBracaloni - when we talked about formatted responses and that it should ideally work in a similar manner to the old search, regarding limit/offset we maybe have two options:

1. Limit/Offset like with old search (with maximum/default limit)
    1. Pro: Exactly like before
    2. Pro: Can eventually retrieve all twins
    3. Con: Repeated pagination is made against all hosts and they all have to re-run the query repeatedly even if no more results are available (which could be expensive)
    4. Con: If wanted to move to a multi-result-per-host streaming model, offset would have to be deprecated (which at some point would be a breaking change)
2. Limit only (with maximum/default limit)
    1. Pro: Clients do not have to paginate (and reason about pagination for multiple hosts with various twin counts)
    2. Pro: Hosts with a small number of results do not have to re-run queries for the purpose of pagination (there is none)
    3. Con: There is a limit to how many twins can be returned which could hide some results (but it could be high)
         - Note: In the old search, jena-text matching also has an internal limit (~10k, I believe).
    5. Pro: Either
        1. offset could be introduced later (no breaking change) **or**
        2. multiple responses per host could be enabled (also arguably not breaking, esp. after Conal's update with host-ID present per twin)

What's your view on this? I'd be tempted to go with (2) simply because of point `2.iv` rather than incurring the penalty of `1.iii`. (We can then still choose to add an offset and/or stream multiple responses per host.)
(As long the proto definition supported `Range` in a same way, I assume.)

Reply:
I would go with solution 1. (as discussed) so it is in line with the existing behavior and it can be addressed as a whole. 

+ it still allows to access all the twins

Reply:
> > Nevertheless, at the moment the scope can either be LOCAL or GLOBAL, but what if there was the possibility to search for Twins within a specific remote Host (or a list of) ?
> 
> @Elleppi - that's a good idea (AKA "targeted search") and in fact has also been suggested for other use-cases too.
> 
> This is independent of search filtering capabilities, so we can treat it as a separate feature. If/when this ends up being something that would be very useful for end-users - we can consider it in the goals. (I'm not saying it's not useful already - just that we have to plan it, taking other needs into account.) (CC @StephanieBracaloni)

@Elleppi a good feedback for https://ioticlabs.atlassian.net/wiki/spaces/DT/pages/982384649/Improve+Usability
I will add it :+1: